src/interfaces/signatures.cpp

```diff
@@ -1,25 +1,30 @@
 #include "signatures.hpp"
+#include "importhelpers.hpp"
 #include <cassert>
 #include <cstring>
 #include <tuple>
+#include <vector>
 
 namespace
 {
-    std::tuple<uint32_t, span<uint8_t>> read_compressed_uint(span<uint8_t> signature)
+    template<typename T, typename = typename std::enable_if<std::is_same<typename std::remove_const<T>::type, uint8_t>::value>::type>
+    std::tuple<uint32_t, span<T>> read_compressed_uint(span<T> signature)
     {
         ULONG value;
         signature = slice(signature, CorSigUncompressData(signature, &value));
         return std::make_tuple(value, signature);
     }
 
-    std::tuple<int32_t, span<uint8_t>> read_compressed_int(span<uint8_t> signature)
+    template<typename T, typename = typename std::enable_if<std::is_same<typename std::remove_const<T>::type, uint8_t>::value>::type>
+    std::tuple<int32_t, span<T>> read_compressed_int(span<T> signature)
     {
-        int value;
+        int value = 0;
```

Comment:
This one is initialized but not the others, is this an oversight?

Reply:
I'll check, but it believe this case required the initialization because of how `CorSigUncompressSignedInt` handles the sign calculations was leading to an "uninitialized value" warning. It's possible that the other cases have the same problem but the compiler only understood this case.